executor/union_scan.go

```diff
@@ -28,12 +29,15 @@ import (
 // DirtyDB stores uncommitted write operations for a transaction.
 // It is stored and retrieved by context.Value and context.SetValue method.
 type DirtyDB struct {
+	sync.Mutex
+
 	// tables is a map whose key is tableID.
 	tables map[int64]*DirtyTable
 }
 
 // GetDirtyTable gets the DirtyTable by id from the DirtyDB.
 func (udb *DirtyDB) GetDirtyTable(tid int64) *DirtyTable {
+	udb.Lock()
```

Comment:
As the comment in `buildUnionScanFromReader` explains, it should be semantically safe even if data race occurs here, so I didn't add synchronization then. How was this panic happened? was TiDB server compiled with `Race Enabled`?